src/Storages/MergeTree/MergePlainMergeTreeTask.cpp

```diff
@@ -32,9 +32,9 @@ bool MergePlainMergeTreeTask::executeStep()
     ProfileEventsScope profile_events_scope(&profile_counters);
 
     /// Make out memory tracker a parent of current thread memory tracker
-    MemoryTrackerThreadSwitcherPtr switcher;
+    ThreadGroupSwitcherPtr switcher;
```

Comment:
Hmm, maybe std::optional is better?

Reply:
Yes, no heap allocations needed here. Will change.